transformations/tests/test_block_index_inject.py

```diff
@@ -0,0 +1,327 @@
+# (C) Copyright 2018- ECMWF.
+# This software is licensed under the terms of the Apache Licence Version 2.0
+# which can be obtained at http://www.apache.org/licenses/LICENSE-2.0.
+# In applying this licence, ECMWF does not waive the privileges and immunities
+# granted to it by virtue of its status as an intergovernmental organisation
+# nor does it submit to any jurisdiction.
+
+from shutil import rmtree
+import pytest
+
+from loki import (
+    Dimension, gettempdir, Scheduler, OMNI, FindNodes, Assignment, FindVariables, CallStatement, Subroutine,
+    Item
+)
+from conftest import available_frontends
+from transformations import BlockViewToFieldViewTransformation, BlockIndexInjectTransformation
+
+@pytest.fixture(scope='module', name='horizontal')
+def fixture_horizontal():
+    return Dimension(name='horizontal', size='nlon', index='jl', bounds=('start', 'end'),
+                     aliases=('nproma',), bounds_aliases=('bnds%start', 'bnds%end'))
+
+
+@pytest.fixture(scope='module', name='blocking')
+def fixture_blocking():
+    return Dimension(name='blocking', size='nb', index='ibl', index_aliases='bnds%kbl')
```

Comment:
Could be achieved with the expression parser ... But probably beyond the scope of this PR.

Reply:
Yes storing `Dimension` members as variables using the expression parser rather than strings would be nicer, and it would makes the checks more robust than string comparisons. Having said that, for the simple use-cases we currently have for `Dimension` objects, it might be overkill 🤷